sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/FirestoreStatefulComponentFactory.java

```diff
@@ -97,9 +100,11 @@ FirestoreStub getFirestoreStub(PipelineOptions options) {
                     .build());
       } else {
         GcpOptions gcpOptions = options.as(GcpOptions.class);
+        String host = System.getenv().getOrDefault(FIRESTORE_HOST_VARIABLE, DEFAULT_FIRESTORE_HOST);
+        firestoreOptions.setHost(host);
```

Comment:
This doesn't seem right. We appear to always be ignoring any host set in the options.

Shouldn't the logic mirror the emulatorHost logic? emulatorHost claims that it prefers 1) value in option then 2) value in env. I expect we should follow the same rules.

Reply:
Revised to get env or default value only if there is no value in options.